### ISSUE TITLE
Regular expressions search option not synchronized between Viewer and Editor

### ISSUE THREAD

**Issue by rohitab** [FarGroup/FarManager#613]
### Far Manager version

3.0.6094.0 x64

### OS version

10.0.22623.1180

### Steps to reproduce

1. Open Editor
2. Press `F7` to open _Search_ dialog
3. Check _Regular expressions_ option
4. Press `Enter` to search
5. Ignore any messages and close Editor
6. Open Viewer
7. Press `F7` to open _Search_ dialog
8. Uncheck _Regular expressions_ option
9. Press `Enter` to search
10. Ignore any messages and close Viewer
11. Open Editor again
12. Press `F7` to open _Search_ dialog

### Expected behavior

_Regular expressions_ option should be unchecked.

### Actual behavior

_Regular expressions_ option is checked. 

Note that all other search options _Case sensitive_, _Whole words_, _Reverse search_ and _Fuzzy search_ are synchronized between the Viewer and Editor. _Regular expressions_ is the only one that is managed separately. It is also the only option that is saved between restarts of _Far_. It appears that these settings are stored in `Editor.SearchRegexp` and `Viewer.SearchRegexp`.

One other minor difference between Viewer and Editor -- in Viewer, when you check _Regular expressions_, _Whole words_ and _Fuzzy search_ are disabled; in Editor they are not.

I am aware that the search is being unified, so most of these issues will most likely be fixed. Decided to report them anyway.

**Comment by MKadaner** [comment]
Thank you for the report, @rohitab, very useful. You are right, the unified search / replace dialog will enable / disable controls as necessary. Since you've tested the PR version of the improved dialog and did not complain, I assume that this functionality works satisfactorily. If I missed something, I will always be able to fix the behavior later, and it will be fixed for all three, Editor, Viewer, and Help.

The _Regex_ option is different. @alabuzhev, do you know why it is handled differently than all other ones? Again, it is easy to fix, but hard to tell who'll shout out.

**Comment by alabuzhev** [comment]
> why it is handled differently than all other ones?

Probably an overlook: regex was added way later than the other options and, as far as I remember, not simultaneously to Viewer / Editor.
I see no reason for it to behave differently.

> in Viewer, when you check Regular expressions, Whole words and Fuzzy search are disabled; in Editor they are not

This is intentional: initially whole words didn't work with regexes anywhere. Relatively recently (in 6013) I added this to Editor (and probably Help too - I think it uses the same function), but not to Viewer, because search was implemented quite differently there and I couldn't be bothered.
Now @MKadaner unified Viewer search with the others to some extent, so it might be easier to add it there as well.

**Comment by MKadaner** [comment]
Please assign this issue to me, I'll adjust saving / restoring Regex option after integrating unified search dialog into Viewer, together with #612.

**Comment by MKadaner** [comment]
Re "Whole words" and "Regex": for now, I removed dependency of the state of "Whole words" checkbox on "Regex" checkbox in #608. It will make sense in Editor (and probably in Help) but will be useless in Viewer until I change Viewer to use common infra for Regex search. I do not want to make this feature conditional in the dialog because "soon" it will unnecessary.

**Comment by MKadaner** [comment]
Oops! It appears that instead of a simple global regex search option, we have these separate config options:
```
Editor.SearchRegexp
Help.HelpSearchRegexp
Viewer.SearchRegexp
```
I'd rather handle regex search option similar to other search options, like _case sensitive_ or _whole words_. However, it means a breaking change for those who have `Auto save setup` set to _on_. I also can keep both, global _and_ config, and sync them in some semi-meaningful way.

@alabuzhev, what do you think? Is it worth trouble, or maybe just forget about config options?

**Comment by alabuzhev** [comment]
@MKadaner I vaguely remember we discussed something like that recently.
I don't mind handling regex the same as CS or WW and see no reason to do it differently.
Breaking change is not a big deal in this case.

**Comment by MKadaner** [comment]
Fixed in 6135 (#678)

**Comment by rohitab** [comment]
Thank you! Verified in version `3.0.6136.0 x64`.
